.github/templates/xla_ci_workflow.yml.j2

```diff
@@ -0,0 +1,457 @@
+{% import 'common.yml.j2' as common %}
```

Comment:
Yes, otherwise, you would need to 1) redundantly set XLA specific configs for all the LINUX workflows (e.g., `XLA_CUDA`) and modify them whenever we change the XLA test settings. What's even more concerning is the conflict of the some config variables (e.g., C++ compiler ABI flag, which is not the case now, but could be. We had a similar conflict between Tensorflow and PyTorch/XLA before). Decoupling should make our lives easier, I feel like.

Reply:
Ah, also this:
```
- name: Calculate docker image tag
        id: calculate-tag
        run: |
          echo "XLA workflow uses pre-built test image at :v0.2"
          DOCKER_TAG=$(git rev-parse HEAD:.circleci/docker)
          echo "DOCKER_TAG=${DOCKER_TAG}" >> "${GITHUB_ENV}"
          echo "DOCKER_IMAGE=${DOCKER_IMAGE_BASE}:v0.2" >> "${GITHUB_ENV}"
          echo "::set-output name=docker_tag::${DOCKER_TAG}"
          echo "::set-output name=docker_image::${DOCKER_IMAGE_BASE}:v0.2"
          ```
 Needed to customize the image tag.

Reply:
I think that adding an entirely new workflow template just for this is going to be a bit cumbersome to maintain. If there are portions of the template that don't necessarily work for XLA we can get around that by adding conditional statements to fix.

If the difference is a couple of conditionals that need to be added I'd prefer that over adding an entirely new workflow altogether.

Reply:
I see, I thought having a separate workflow template might be leaner and isolate XLA testing issues from the others (similar to mac, android testing, etc.) -- 1) XLA related workflow template changes would not affect/break others and 2) adding conditionals... for XLA testing setup might string/clutter the linux template down the road. 

Having said that, if a separate XLA template is more burdensome, I would go with your guidance using conditionals. Shall we?